DotNetCasClient/Security/AssertionRoleProvider.cs

```diff
@@ -89,36 +89,52 @@ public override string[] FindUsersInRole(string roleName, string usernameToMatch
             throw new NotImplementedException();
         }
 
-        /// <summary>
-        /// Get all the roles we know about from the current user's CAS assertion.
-        /// This operation is synonymous with <see cref="GetRolesForUser"/>.
-        /// </summary>
-        /// <returns>List of all roles belonging to current user.</returns>
-        public override string[] GetAllRoles()
-        {
-            IList<string> roles = GetCurrentUserRoles();
-            if (roles is Array)
-            {
-                return (string[])roles;
-            }
-            string[] roleArray = new string[roles.Count];
-            for (int i = 0; i < roles.Count; i++)
-            {
-                roleArray[i] = roles[i];
-            }
-            return roleArray;
-        }
-
-        public override string[] GetRolesForUser(string username)
-        {
-            if (CasAuthentication.CurrentPrincipal.Identity.Name != username)
-            {
-                throw new ProviderException("Cannot fetch roles for user other than that of current context.");
-            }
-            return GetAllRoles();
-        }
-
-        public override string[] GetUsersInRole(string roleName)
+		/// <summary>
+		/// Gets a list of all the roles for the configured applicationName.
+		/// </summary>
+		/// <returns>A string array containing the names of all the roles stored in the data
+		/// source for the configured applicationName.</returns>
+		/// <remarks>This method will always throw a <see cref="NotImplementedException"/> as the
+		/// CAS client is not able to retrieve a list of all roles from the CAS server.</remarks>

```

Comment:
What are your thoughts on putting an Obsolete attribute on this specific method?

Reply:
Uh, dumb line comment thing.  It should be showing you I'm referring to the GetAllRoles() method.

Reply:
The anal retentive characteristic in me doesn't want to use it because it's not 100% correct, but I think it's good enough, especially since we can configure the attribute to cause an error to be raised whenever the method is referenced. For the message I would suggest something like "Method is not supported by this provider."

Additionally, if you want to proceed with doing this, I think a new issue should be created to review all the code and do this in similar places so that the entire assembly is consistent.

Reply:
On second though this one might be the best option: https://docs.microsoft.com/en-us/dotnet/api/system.notsupportedexception?view=netframework-4.8

Right?

Reply:
Yep!

"The exception that is thrown when an invoked method is not supported"
    is more correct/clear than
"The exception that is thrown when a requested method or operation is not implemented"

I'll submit a PR soon to fix this.